File: leetcode/js/set-mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function (nums) {
  const counts = Array(nums.length + 1).fill(0);

  for (const num of nums) {
    counts[num] += 1;
  }

  return [counts.indexOf(2), counts.slice(1).indexOf(0) + 1];
};

File: leetcode/rust/set-mismatch.rs
impl Solution {
    pub fn find_error_nums(nums: Vec<i32>) -> Vec<i32> {
        use std::collections::HashSet;

        let mut result = Vec::new();
        let mut set = HashSet::new();

        for num in nums.iter() {
            if set.contains(num) {
                result.push(*num);
            }

            set.insert(*num);
        }

        for num in 1..=(nums.len() as i32) {
            if !set.contains(&num) {
                result.push(num);
                break;
            }
        }

        result
    }
}
